SU2_CFD/src/output/COutput.cpp

```diff
@@ -0,0 +1,1857 @@
+      }
+    }
+  }
+  
+  if (config->GetTime_Domain() && config->GetDirectDiff()){
+    for (unsigned short iField = 0; iField < historyOutput_List.size(); iField++){
+      HistoryOutputField &currentField = historyOutput_Map[historyOutput_List[iField]];
+      if (currentField.fieldType == TYPE_COEFFICIENT){
+        AddHistoryOutput("D_TAVG_" + historyOutput_List[iField], "dtavg[" + currentField.fieldName + "]",
+                         currentField.screenFormat, "D_TAVG_" + currentField.outputGroup, 
+                         "Derivative of the time averaged value (DIRECT_DIFF=YES)", TYPE_AUTO_COEFFICIENT);  
+      }
+    }
+  }
+  
+  if (historyOutput_Map[convField].fieldType == TYPE_COEFFICIENT){
+    AddHistoryOutput("CAUCHY", "C["  + historyOutput_Map[convField].fieldName + "]", FORMAT_SCIENTIFIC, "CAUCHY",
+                     "Cauchy residual value of field set with CONV_FIELD." ,TYPE_AUTO_COEFFICIENT);
+  }
+
+}
+
+bool COutput::WriteScreen_Header(CConfig *config) {  
+  
+  unsigned long RestartIter = 0;
+  
+  if (config->GetRestart() && config->GetTime_Domain()){
+    RestartIter = config->GetRestart_Iter();
+  }
+  
+  unsigned long ScreenWrt_Freq_Inner = config->GetScreen_Wrt_Freq(2);
+  unsigned long ScreenWrt_Freq_Outer = config->GetScreen_Wrt_Freq(1);
+  unsigned long ScreenWrt_Freq_Time  = config->GetScreen_Wrt_Freq(0);
+  
+  /*--- Always print header if it is forced ---*/
+  
+  if (headerNeeded){
+    headerNeeded = false;
+    return true;
+  }
+  
+  /*--- Header is always disabled for multizone problems unless explicitely requested --- */
+  
+  if (config->GetMultizone_Problem() && !config->GetWrt_ZoneConv()){
+    return false;
+  }
+
+  /* --- Always print header in the first iteration --- */
+  
+  if ((curInnerIter == 0) && 
+      (curOuterIter == 0) && 
+      (curTimeIter == RestartIter)){
+    return true;
+  }
+  
+  if (!PrintOutput(curTimeIter, ScreenWrt_Freq_Time)&& 
+      !(curTimeIter == config->GetnTime_Iter() - 1)){
+    return false;
+  }
+   
+  /*--- If there is no inner or outer iteration, don't print header ---*/
+  if (ScreenWrt_Freq_Outer == 0 && ScreenWrt_Freq_Inner == 0){
+    return false;
+  }
+  
+  /*--- Print header if we are at the first inner iteration ---*/
+  
+  if (curInnerIter == 0){
+    return true;
+  }
+  
+  return false;
+}
+
+bool COutput::WriteScreen_Output(CConfig *config) {
+  
+  unsigned long ScreenWrt_Freq_Inner = config->GetScreen_Wrt_Freq(2);
+  unsigned long ScreenWrt_Freq_Outer = config->GetScreen_Wrt_Freq(1);
+  unsigned long ScreenWrt_Freq_Time  = config->GetScreen_Wrt_Freq(0);    
+  
+  if (config->GetMultizone_Problem() && !config->GetWrt_ZoneConv()){
+    
+    return false;
+    
+  }
+  
+  /*--- Check if screen output should be written --- */
+  
+  if (!PrintOutput(curTimeIter, ScreenWrt_Freq_Time)&& 
+      !(curTimeIter == config->GetnTime_Iter() - 1)){
+    
+    return false;
+    
+  }
+  
+  if (convergence) {return true;}
```

Comment:
this check is not needed

Reply:
The rest of the function should not be executed if convergence is true